### PR TITLE
Fix(post-processing) Division by zero fix when calculating metrics

### PR DESCRIPTION
**Pull request by Vince-janv**
## Description
When calculating some metrics for the bcl2fastq flow we allow some fields to be zero and then divide by them which raises `ZeroDivisionError`.

This PR makes it so that the functions returns 0 in those cases, and also adds tests to for the edge cases.

### Added

- Tests for cases where `total_yield` is 0

### Changed

- `calculate_q30_bases_percentage` and `calculate_average_quality_score` return 0 if `total_yield` is 0

### This [version](https://semver.org/) is a

- [x] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

**Review by islean — APPROVED** [review]
Looks good! 🔥

File: tests/apps/sequencing_metrics_parser/test_sequencing_metrics_calculator.py [hunk 1]
```diff
@@ -1,32 +1,40 @@
...
+
 from cg.apps.sequencing_metrics_parser.sequencing_metrics_calculator import (
     calculate_q30_bases_percentage,
     calculate_average_quality_score,
 )
 
 
-def test_calculate_bases_with_q30_percentage():
+@pytest.mark.parametrize(
+    "q30_yield, total_yield, expected_result", [(100, 200, 100 / 200 * 100), (0, 0, 0)]
```
[the remark this answers is not in the document]
Comment (Vince-janv) [reply tests/apps/sequencing_metrics_parser/test_sequencing_metrics_calculator.py:10]: Python is smart enough to know that `100/200*100 == (100/200)*100` That was the order in the functions it was testing, so that's why i kept it. Would you prefer parentheses?

[the remark this answers is not in the document]
Comment (islean) [reply tests/apps/sequencing_metrics_parser/test_sequencing_metrics_calculator.py:10]: I always like the added clarity, but feel free to resolve without changing it if it is not worth the time.
